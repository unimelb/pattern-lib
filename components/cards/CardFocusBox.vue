<template>
  <FocusWrapper
    :semi-opaque="semiOpaque"
    :color="color"
    :class="classes"
    padded>
    <!-- eslint-disable vue/no-v-html -->
    <svg
      v-if="icon"
      class="card--focus-box__icon"
      viewBox="20 0 150 150"
      width="100%"
      height="100%"
      v-html="icon" />
    <slot />
    <!-- eslint-enable vue/no-v-html -->
  </FocusWrapper>
</template>

<script>
import FocusWrapper from '../focus-wrapper/FocusWrapper.vue';

export default {
  name: 'CardFocusBox',
  components: { FocusWrapper },
  props: {
    icon: {
      type: String,
      default: '',
    },
    title: {
      type: String,
      default: 'Test title',
    },
    color: {
      type: String,
      default: '',
      validator: (color) => ['', 'navy', 'teal', 'yellow', 'emerald', 'orange', 'green', 'purple', 'pink'].includes(color),
    },
    href: {
      type: String,
      default: '#',
    },
    vertical: {
      type: Boolean,
      default: false,
    },
    excerpt: {
      type: String,
      default:
        'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Soluta quae doloribus, aliquid modi? Blanditiis quod, ullam aliquam provident, accusamus neque aut necessitatibus accusantium maxime voluptates sit consequatur illum tempora magnam.',
    },
    semiOpaque: {
      type: Boolean,
      default: false,
    },
  },
  computed: {
    classes() {
      return {
        'card--focus-box-vert': this.vertical,
      };
    },
  },
};
</script>
