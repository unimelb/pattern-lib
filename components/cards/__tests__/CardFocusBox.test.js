import {
  shallow, mount,
} from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import CardFocusBox from '../CardFocusBox.vue';

expect.extend(toHaveNoViolations);

describe('CardFocusBox', () => {
  it('should match snapshot', () => {
    const result = shallow(CardFocusBox).element;
    expect(result).toMatchSnapshot();
  });

  it('Component throws no accessibility violations', (done) => {
    const html = mount(CardFocusBox).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
