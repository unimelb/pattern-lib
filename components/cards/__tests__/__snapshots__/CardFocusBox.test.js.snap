// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardFocusBox should match snapshot 1`] = `<!---->`;
